Dockerfile

```diff
@@ -18,9 +18,7 @@ COPY --from=builder /app .
 RUN pip install --upgrade pip && \
     pip install -r requirements.txt
```

Comment:
It's a good practice to separate the installation of dependencies from the copying of application code to leverage Docker's caching mechanism. Consider copying only the `requirements.txt` file first, running `pip install`, and then copying the rest of the application code.